ydb/library/yql/minikql/comp_nodes/mkql_grace_join_imp.cpp

```diff
@@ -417,14 +416,15 @@ void TTable::Join( TTable & t1, TTable & t2, EJoinKind joinKind, bool hasMoreLef
             auto slotIt = joinSlots.begin() + slotNum * slotSize;
             while (*slotIt != 0 && slotIt != joinSlots.end())
             {
+
                 bool matchFound = false;
-                if (keysValSize <= slotSize && !JoinTable1->NumberOfKeyIColumns ) {
+                if (((keysValSize - nullsSize1) <= (slotSize - nullsSize2)) && !JoinTable1->NumberOfKeyIColumns ) {
                     if (std::equal(it1 + keyIntOffset1, it1 + keysValSize, slotIt + keyIntOffset2)) {
                         matchFound = true;
                     }
                 }
 
-                if (keysValSize > slotSize && !JoinTable1->NumberOfKeyIColumns ) {
+                if (((keysValSize - nullsSize1) > (slotSize - nullsSize2)) && !JoinTable1->NumberOfKeyIColumns ) {
```

Comment:
better
```
keysValSize + nullsSize2 > slotSize + nullsSize1
```

Reply:
keysValSize is related to probe table 1 (nullsSize1).  slotSize is related to hash table (nullsSize2).  So it is more natural.